src/main/webapp/assets/js/foodMeetings.js

```diff
@@ -94,24 +85,60 @@ $(document).ready(function () {
 		}
 	});
 
-	$("#ImageLinkId").change(function (event) {
-
-		var imageLink = $("#ImageLinkId").val();
-
-		$("#ImageCard").attr("src",imageLink);
-	});
-
 	$("#AddNewMeeting").click(function (event) {
 		
 		$("#NewMeetingCard").show();
-		foodMeetings.masonry({
-			itemSelector: '.grid-item',
-			columnWidth: 50
-		});
+		reorganizeCards();
 	});
 
 	$("#cancelCreateMeeting").click(function (event) {
 		location.reload(); 
 	});
+	
+	var onModalHide = function() {
+		var imageLink = $("#imageLink").val();
+		$("#ImageCard").attr("src", imageLink);
+		reorganizeCards();
+		
+	};
+	
+	$("#ImageCard").click(function () {
+		
 
-});
+		$('#modal1').openModal({
+			complete : onModalHide
+		});
+
+		$('#modal1').load('searchImage',function(data){
+			$("#imageLink").change(function (event) {
+				var imageLink = $("#imageLink").val();
+				$("#imageCard").attr("src",imageLink);
+			});
+			
+			$(".image-link").click(function () {
+				
+				$("#imageCard").attr("src", $(this).data("imageLink"));
+				$("#imageLink").val($(this).data("imageLink"));
+				
+			});
+
+			var imagesLinks = $('.image-links');
+
+			imagesLinks.imagesLoaded()
+				.done(function(){
+					imagesLinks.masonry({
+						itemSelector: '.grid-item',
+						columnWidth: 50
+					});
+				});
+		});
+	});
+	
+	function reorganizeCards() {
```

Comment:
Good to have this in a separate method!.  Following masonry speech, I would prefer call this method "layoutFoodMeetingCards" if you agree. http://masonry.desandro.com/methods.html#layout-masonry